components/textbook-demo/TextbookDemoHeader.vue

```diff
@@ -125,9 +125,8 @@ export default class TextbookDemoHeader extends (Vue as VueConstructor<VueCompon
     margin-top: $layout-03;
 
     &_fixed {
-      position: fixed;
+      position: fixed !important;
```

Comment:
I was trying to avoid the use of this fixed parameter in order to also avoid the `!important` annotation. I found [here](https://www.geekabyte.io/2014/03/when-css-position-sticky-doesnt-work.html) in a comment: 

> once a sticky element's parent goes out of view, the element will no longer stick.

It is possible to fix but probably not worth it.
To avoid the `!important` you can use a selector with higher specificity: `&_fixed.content-menu-section`

Reply:
I'm all for avoiding `!important` when possible, but here I thought it might be a cleaner and safer approach.

By adding specificity via the `.content-menu-section` class, we depend on said class name, but this component isn't aware of that class. The clases in the `TextbookDemoContentMenuSection` component –including (`.content-menu-section`)– aren't exposed in purpose. We would be just kind of using an unintended side-effect here IMO. That class could change down the road making this solution fail silently for instance.

Reply:
`sticky`, as you point out, makes the element stick to the nearest ancestor and containing block (https://developer.mozilla.org/en-US/docs/Web/CSS/position#values).

I considered some ideas about adjusting the order or the elements to have the "sticky navbar" rendered higher in the DOM tree, but they were much more complex and didn't really provide any significant benefit that out-wages the downside of said complexity.

Besides being generally agains `!important`, it's a feature that we still can use, as long as we are careful and don't overuse it.

Reply:
You are right about the "fail silently" reasoning, but then I think that the `!important` could do something similar in some cases. Also, if we want to make a variant of this class we should overwrite again the `!important` using specificity.

The only permitted use case of `!important` in [Mozilla](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity) is to overwrite external CSS such as bootstrap. If we externalize this component, could be a use case. But we are working with `v-deep` in some cases to prevent the use of `!important` with the carbon components, so I think that we should follow the same pattern in all the code.

Maybe this is a good discussion with all the dev team to see edge cases for `!important`. But at this moment, I think is better to follow our current patterns.

Reply:
There is a very good argument about keeping the component independent in terms of positioning and external geometry as mentioned in BEM's documentation (https://en.bem.info/methodology/faq/#why-is-external-geometry-and-positioning-set-via-the-parent-block), as discussed with @delapuente offline.

I've corrected this with https://github.com/Qiskit/qiskit.org/pull/1901/commits/cfe8be0bf387cb5a01b2ee80742b10c4605615d0.